Mediapipe.Net/Framework/Format/ImageFrame.cs

```diff
@@ -0,0 +1,259 @@
+// Copyright (c) homuler and Vignette
+// This file is part of MediaPipe.NET.
+// MediaPipe.NET is licensed under the MIT License. See LICENSE for details.
+
+using System;
+using Mediapipe.Net.Core;
+using Mediapipe.Net.Native;
+
+namespace Mediapipe.Net.Framework.Format
+{
+    public class ImageFrame : MpResourceHandle
+    {
+        public static readonly uint DefaultAlignmentBoundary = 16;
+        public static readonly uint GlDefaultAlignmentBoundary = 4;
+
+        public delegate void Deleter(IntPtr ptr);
+
+        public ImageFrame() : base()
+        {
+            UnsafeNativeMethods.mp_ImageFrame__(out var ptr).Assert();
+            Ptr = ptr;
+        }
+
+        public ImageFrame(IntPtr imageFramePtr, bool isOwner = true) : base(imageFramePtr, isOwner) { }
+
+        public ImageFrame(ImageFormat format, int width, int height) : this(format, width, height, DefaultAlignmentBoundary) { }
+
+        public ImageFrame(ImageFormat format, int width, int height, uint alignmentBoundary) : base()
+        {
+            UnsafeNativeMethods.mp_ImageFrame__ui_i_i_ui(format, width, height, alignmentBoundary, out var ptr).Assert();
+            Ptr = ptr;
+        }
+
+        // NOTE: This byte* was a NativeArray<byte> from Unity. See if it adapts.
+        unsafe public ImageFrame(ImageFormat format, int width, int height, int widthStep, byte* pixelData)
+        {
+            unsafe
+            {
+                UnsafeNativeMethods.mp_ImageFrame__ui_i_i_i_Pui8_PF(
+                  format, width, height, widthStep,
+                  (IntPtr)pixelData,
+                  releasePixelData,
+                  out var ptr
+                ).Assert();
+                Ptr = ptr;
+            }
+        }
```

Comment:
I used `byte*` here but don't know if it's appropriate

Reply:
Check the return type in the native side so you know how we can handle this. CLR types correspond to C++11/C99 data types so check them properly if you can.

Reply:
Oh true, I didn't think of checking the native side.

Reply:
This is the signature:
```c++
MP_CAPI(MpReturnCode) mp_ImageFrame__ui_i_i_i_Pui8_PF(mediapipe::ImageFormat::Format format, int width, int height, int width_step, uint8* pixel_data,
                                                      Deleter* deleter, mediapipe::ImageFrame** image_frame_out);
```
`byte` directly corresponds to `uint8_t`, so I think we're good

Reply:
I would be certain if the type in this signature was `uint8_t` instead of `uint8`, but honestly there's probably no significant difference.